### PR TITLE
fix: refactor exceptions into seperate files

### PR DESCRIPTION
**Pull request by pbr0ck3r**
Please read [CONTRIBUTING.md](CONTRIBUTING.md) for additional information on contributing to this repository!

## What this PR does / why we need it
- initial take on how we could refactor exceptions into their own files using JSON.

## PR Checklist

- [ ] This PR adds K8s exceptions (false positives)
- [x] This PR adds new code
- [ ] This PR includes test for any new code

## Github Issue

https://github.com/yonahd/kor/issues/253

### CODE REVIEW

File: pkg/kor/kor.go [hunk 1]
```diff
@@ -27,6 +28,11 @@ type IncludeExcludeLists struct {
 	ExcludeListStr string
 }
 
+type Config struct {
```
Comment (yonahd) [inline pkg/kor/kor.go:31]: Interesting. Did you have something in mind to support using this struct?

File: pkg/kor/exceptions/storageclasses/config.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (yonahd) [inline pkg/kor/exceptions/storageclasses/config.json:1]: Nit: maybe storageclasses.json?

**Review by yonahd — COMMENTED** [review]
Looks simple and straightforward.
A couple points here:

- Maybe we can add a field of source which would be more for documentation e.g. GKE or minikube
- The folder name we call it exceptions, trying to think if there is a better name for the false positives

File: pkg/kor/exceptions/storageclasses/config.json (hunk 2 above)
Comment (pbr0ck3r) [reply pkg/kor/exceptions/storageclasses/config.json:1]: Happy to make this change!

File: pkg/kor/kor.go (hunk 1 above)
Comment (pbr0ck3r) [reply pkg/kor/kor.go:31]: It is used by `unmarshalConfig`. But happy to refactor that into not using it. The idea is that this sets us up in a way we could break out exceptions into individual cloud if we want in the future. Then we could adjust the code to do say GKE only (if you pass in a flag) or all by default.

Comment (pbr0ck3r) [reply pkg/kor/kor.go:31]: If we don't use a struct like this, then we will have to have each embedded file be its own struct in each of their individual files to unmarshal the data.

```
type Config struct {
	ExceptionStorageClasses []ExceptionResource `json:"exceptionStorageClasses"`
}

type ExceptionClusterRoles struct {
	ExceptionClusterRoles []ExceptionResource `json:"exceptionClusterRoles"`
}
```

Which will then require us to refactor the `unmarshalConfig` helper.

The other idea would be do away with JSON files altogether and just  create another Go package called `exceptions` and we move everything, i.e. `pkg/exceptions/exceptionStorageClasses.go`, `pkg/exceptions/exceptionClusterRoles.go`

Comment (yonahd) [reply pkg/kor/kor.go:31]: > It is used by `unmarshalConfig`. But happy to refactor that into not using it. The idea is that this sets us up in a way we could break out exceptions into individual cloud if we want in the future. Then we could adjust the code to do say GKE only (if you pass in a flag) or all by default.

I actually did not think of that. Really solid idea!

Comment (pbr0ck3r) [reply pkg/kor/kor.go:31]: I pushed up some more code to show how the `Config` struct will look in the future.

Comment (yonahd) [reply pkg/kor/kor.go:31]: Looks good to me. Go for it

**Review by yonahd — APPROVED** [review]
Lgtm
Thank you!

**Pull request merged**